include/odp/api/spec/packet.h

```diff
@@ -2140,6 +2141,57 @@ uint32_t odp_packet_payload_offset(odp_packet_t pkt);
  */
 int odp_packet_payload_offset_set(odp_packet_t pkt, uint32_t offset);
 
+/**
+ * Set proto stats object.
+ *
+ * Associate a proto stats object to a packet. This enables updating protocol
+ * stats of the packet in the given object after packet is enqueued to queue
+ * and reaches a terminal state i.e either transmit or drop.
```

Comment:
The "enqueued to queue" part is misleading since the stats are not updated in all ODP queues but only when the packet is leaving the system through a pktout.

It would also be good the specify what happens when a packet gets fragmented (e.g. in inline IPsec output) or fragmented/segmented (through the LSO mechanism).

I would rephrase:

Associate a packet with a proto stats object. The statistics enabled in the proto stats object are updated for the packet in packet output when the packet is transmitted or dropped. The statistics update is done as the last step of output processing after possible packet transformations (e.g. fragmentation, IPsec) that may occur. If a packet is fragmented or segmented to multiple packets as part of output processing, all the resulting packets inherit the proto stats object association from the original packet.